services/cd-service/pkg/repository/transformer_test.go

```diff
@@ -793,6 +793,41 @@ func TestRbacTransformerTest(t *testing.T) {
 		Transformers  []Transformer
 		ExpectedError string
 	}{
+		{
+			Name: "able to create application version with permissions policy",
+			Transformers: []Transformer{
+				&CreateEnvironment{
+					Environment: "acceptance",
+					Config:      config.EnvironmentConfig{Upstream: &config.EnvironmentConfigUpstream{Environment: envAcceptance, Latest: false}},
```

Comment:
you can't have the same env as upstream

Reply:
Fixed the test.

Reply:
Following our discussion I also added the Authentication to `DeployApplicationVersion` that was missing. Added a test for covering this case as well.